pkg/reconciler/reconciler_test.go

```diff
@@ -1444,6 +1463,67 @@ var _ = Describe("Reconciler", func() {
 			Expect(controllerSetupCalled).To(BeTrue())
 		})
 	})
+
+	var _ = Describe("Test label selector for two reconcilers", func() {
+		var (
+			mgr              manager.Manager
+			firstReconciler  *Reconciler
+			secondReconciler *Reconciler
```

Comment:
Were you able to reproduce the wrong reconciles with this test?
Could you mark this as a regression test in the description or in a comment?
I.e.
`Test ensures that dependend, not-labeled, watches like secrets do not trigger a reconcile on objects which should not be reconciled based on the label selector.`

Reply:
The unit test in this case does not test the issue, so it can be removed as it calls `Reconcile` directly. Instead of testing the reconcile is it possible to verify that all watches contain the necessary predicates?
Otherwise a kuttl or envtest test could test this bug.

Reply:
Changed tests with new fix implementation